### PR TITLE
storj-sdk base code

### PR DESCRIPTION
**Pull request by egonelbre**
_Note the arguments might not still be correct and/or different binaries require waiting for different parts to come online rather than failing immediately. But should be good for starting the review._

Base code with few commands:

* [x] `STORJ_NETWORK_DIR=local` to set the directory for all the peers (or use `--config-dir` argument).
* [x] `storj-sdk network setup` to setup a local network.
* [x] `storj-sdk network run` to run network that was setup.
* [x] `storj-sdk network destroy` to delete the network directory.
* [x] `storj-sdk network test <command>` to run `<command>` using network in `local` directory. The satellite addresses, identities are specified by environment flags like `SATELLITE0_ADDR`.
* [x] `storj-sdk test <command>` to run in-memory network with `<command>`. The satellite addresses, identities are specified by environment flags like `SATELLITE0_ADDR`, `SATELLITE0_ID`.

### CODE REVIEW

File: cmd/storj-sdk/network.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+	)
+
+	defaultSatellite := net.JoinHostPort(host, strconv.Itoa(satellitePort+0))
+
+	arguments := func(name, command, addr string, rest ...string) []string {
+		return append([]string{
+			"--log.level", "debug",
+			"--config-dir", ".",
+			command,
+			"--identity.server.address", addr,
```
Comment (jtolds) [inline cmd/storj-sdk/network.go]: my last merged pr changed this to --server.address, and then an upcoming pr changes this to --server.public-address, fyi

File: cmd/storj-sdk/network.go [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+			"--kademlia.bootstrap-addr", defaultSatellite,
+		)
+	}
+
+	gatewayArguments := func(name, command string, addr string, rest ...string) []string {
+		return append([]string{
+			"--log.level", "debug",
+			"--config-dir", ".",
+			command,
+			"--identity.server.address", addr,
```
Comment (jtolds) [inline cmd/storj-sdk/network.go]: this should just become --server.address after both prs

File: cmd/storj-sdk/network.go [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+		process.Info.Address = net.JoinHostPort(host, strconv.Itoa(storageNodePort+i))
+
+		process.Arguments["setup"] = arguments(name, "setup", process.Info.Address,
+			"--piecestore.agreementsender.overlay-addr", defaultSatellite,
+		)
+		process.Arguments["run"] = arguments(name, "run", process.Info.Address,
+			"--piecestore.agreementsender.overlay-addr", defaultSatellite,
+			"--kademlia.bootstrap-addr", defaultSatellite,
+			"--kademlia.farmer.email", fmt.Sprintf("storage%d@example.com", i),
+			"--kademlia.farmer.wallet", "0x0123456789012345678901234567890123456789",
```
Comment (jtolds) [inline cmd/storj-sdk/network.go]: these changed from farmer to operator.

bummer this is kind of brittle, but i don't know what else to do. we should make sure running these codepaths are part of builds

File: cmd/storj-sdk/prefix.go [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+	root    *prefixWriter
+	maxline int
+
+	mu  sync.Mutex
+	len int
+	dst io.Writer
+}
+
+// NewPrefixWriter creates a writer than can prefix all lines written to it.
+func NewPrefixWriter(defaultPrefix string, dst io.Writer) *PrefixWriter {
```
Comment (jtolds) [inline cmd/storj-sdk/prefix.go]: neat

File: cmd/storj-sdk/network.go (hunk 3 above)
Comment (egonelbre) [reply cmd/storj-sdk/network.go]: we have some things still to do before we can start using it, but once we have it running we can switch captplanet integration test for storj-sdk test.

**Pull request merged**